graph/dijkstra.cpp

```diff
@@ -1,52 +1,177 @@
-#include <cstdio>
+/**
+ * @file
+ * @brief [Graph Dijkstras Shortest Path Algorithm
+ * (Dijkstra's Shortest Path)]
+ * (https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm)
+ *
+ * @author [Ayaan Khan](http://github.com/ayaankhan98)
+ *
+ * @details
+ * Dijkstra's Algorithm is used to find the shortest path from a source
+ * vertex to all other reachable vertex in the graph.
+ * The algorithm initially assumes all the nodes are unreachable from the
+ * given source vertex so we mark the distances of all vertices as INF
+ * (infinity) from source vertex (INF / infinity denotes unable to reach).
+ *
+ * in similar fashion with BFS we assume the distance of source vertex as 0
+ * and pushes the vertex in a priority queue with it's distance.
+ * we maintain the priority queue as a min heap so that we can get the
+ * minimum element at the top of heap
+ *
+ * Basically what we do in this algorithm is that we try to minimize the
+ * distances of all the reachable vertices from the current vertex, look
+ * at the code below to understand in better way.
+ *
+ */
+#include <cassert>
 #include <iostream>
 #include <queue>
+#include <utility>
 #include <vector>
-using namespace std;
-#define INF 10000010
-vector<pair<int, int>> graph[5 * 100001];
-int dis[5 * 100001];
-int dij(vector<pair<int, int>> *v, int s, int *dis) {
-    priority_queue<pair<int, int>, vector<pair<int, int>>,
-                   greater<pair<int, int>>>
+
+constexpr long long INF = 1000000000;
```

Comment:
you can use `std::numeric_limits<long long>::max()` for infinity

Reply:
okay thanks, see for further improvements and let me know